driver-async/src/main/com/mongodb/async/client/gridfs/GridFSDownloadStreamImpl.java

```diff
@@ -144,6 +144,48 @@ public void onResult(final Integer result, final Throwable t) {
         });
     }
 
+    @Override
+    public void skip(final long bytesToSkip, final SingleResultCallback<Long> callback) {
+        notNull("callback", callback);
+        final SingleResultCallback<Long> errHandlingCallback = errorHandlingCallback(callback, LOGGER);
+        if (checkClosed()) {
+            errHandlingCallback.onResult(null, null);
+        } else if (!hasFileInfo()) {
+            getGridFSFile(new SingleResultCallback<GridFSFile>() {
+                @Override
+                public void onResult(final GridFSFile result, final Throwable t) {
+                    if (t != null) {
+                        errHandlingCallback.onResult(null, t);
+                    } else {
+                        skip(bytesToSkip, errHandlingCallback);
+                    }
+                }
+            });
+        } else if (bytesToSkip <= 0) {
+            callback.onResult(0L, null);
+        } else {
+            long skippedPosition = currentPosition + bytesToSkip;
```

Comment:
This is the same logic as the sync driver.

Reply:
Ack